AttributeError: 'Extension' object has no attribute 'hpy_abi'

My goal was to try piconumpy with a PyPy nightly build after the merge of the hpy-0.0.2 branch but I get other problems :-)

I realized that one can't compile piconumpy with hpy master. It seems to be related to the fact that we also build in piconumpy the old C extension using only the CPython API.

```
$ python setup.py develop                                                                                                                     
running develop
running egg_info
writing piconumpy.egg-info/PKG-INFO
writing dependency_links to piconumpy.egg-info/dependency_links.txt
writing requirements to piconumpy.egg-info/requires.txt
writing top-level names to piconumpy.egg-info/top_level.txt
writing manifest file 'piconumpy.egg-info/SOURCES.txt'
running build_ext
building 'piconumpy._piconumpy_cpython_capi' extension
creating build
creating build/temp.linux-x86_64-3.8
creating build/temp.linux-x86_64-3.8/piconumpy
gcc -pthread -Wno-unused-result -Wsign-compare -DNDEBUG -g -fwrapv -O3 -Wall -fPIC -I/home/pierre/.pyenv/versions/3.8.2/include/python3.8 -c piconumpy/_piconumpy_cpython_capi.c -o build/temp.linux-x86_64-3.8/piconumpy/_piconumpy_cpython_capi.o -Wfatal-errors -Werror
Traceback (most recent call last):
  File "setup.py", line 4, in <module>
    setup(
  File "/home/pierre/.pyenv/versions/3.8.2/lib/python3.8/site-packages/setuptools/__init__.py", line 153, in setup
    return distutils.core.setup(**attrs)
  File "/home/pierre/.pyenv/versions/3.8.2/lib/python3.8/distutils/core.py", line 148, in setup
    dist.run_commands()
  File "/home/pierre/.pyenv/versions/3.8.2/lib/python3.8/distutils/dist.py", line 966, in run_commands
    self.run_command(cmd)
  File "/home/pierre/.pyenv/versions/3.8.2/lib/python3.8/distutils/dist.py", line 985, in run_command
    cmd_obj.run()
  File "/home/pierre/.pyenv/versions/3.8.2/lib/python3.8/site-packages/setuptools/command/develop.py", line 34, in run
    self.install_for_development()
  File "/home/pierre/.pyenv/versions/3.8.2/lib/python3.8/site-packages/setuptools/command/develop.py", line 136, in install_for_development
    self.run_command('build_ext')
  File "/home/pierre/.pyenv/versions/3.8.2/lib/python3.8/distutils/cmd.py", line 313, in run_command
    self.distribution.run_command(command)
  File "/home/pierre/.pyenv/versions/3.8.2/lib/python3.8/distutils/dist.py", line 985, in run_command
    cmd_obj.run()
  File "/home/pierre/.pyenv/versions/3.8.2/lib/python3.8/site-packages/setuptools/command/build_ext.py", line 79, in run
    _build_ext.run(self)
  File "/home/pierre/.pyenv/versions/3.8.2/lib/python3.8/distutils/command/build_ext.py", line 340, in run
    self.build_extensions()
  File "/home/pierre/.pyenv/versions/3.8.2/lib/python3.8/distutils/command/build_ext.py", line 449, in build_extensions
    self._build_extensions_serial()
  File "/home/pierre/.pyenv/versions/3.8.2/lib/python3.8/distutils/command/build_ext.py", line 474, in _build_extensions_serial
    self.build_extension(ext)
  File "/home/pierre/.pyenv/versions/3.8.2/lib/python3.8/site-packages/setuptools/command/build_ext.py", line 196, in build_extension
    _build_ext.build_extension(self, ext)
  File "/home/pierre/.pyenv/versions/3.8.2/lib/python3.8/distutils/command/build_ext.py", line 556, in build_extension
    export_symbols=self.get_export_symbols(ext),
  File "/home/pierre/Dev/hpy/hpy/devel/__init__.py", line 296, in get_export_symbols
    if ext.hpy_abi == 'universal':
AttributeError: 'Extension' object has no attribute 'hpy_abi'
```

The https://github.com/hpyproject/piconumpy/branch/hpy branch builds for me with hpy master (3491ae617f601af) but fails on hpy 0.0.2 a different error to yours:
```shell
$ python setup.py develop
...
piconumpy/_piconumpy_hpy.c:9:1: error: return type defaults to ‘int’ [-Werror=implicit-int]
    9 | HPyType_HELPERS(ArrayObject)
      | ^~~~~~~~~~~~~~~
compilation terminated due to -Wfatal-errors.
```
I'm not sure what piconumpy branch you are using, or what operating system, etc but it would be good to reproduce the error first.

The `HPyType_HELPERS` are a new HPy feature added post 0.0.2.

--- a/hpy/devel/__init__.py
+++ b/hpy/devel/__init__.py
@@ -293,6 +293,6 @@ class build_hpy_ext_mixin:
             module "HPyInit_" function.
         """
         exports = self._base_build_ext.get_export_symbols(self, ext)
-        if ext.hpy_abi == 'universal':
+        if hasattr(ext, "hpy_abi") and ext.hpy_abi == 'universal':
             exports = [re.sub(r"^PyInit_", "HPyInit_", name) for name in exports]
         return exports

